### PR TITLE
feat: executable plugin

### PR DESCRIPTION
**Pull request by FroyaTheHen**
commit-id:eefec831

---

**Stack**:
- #1840
- #1839 ⬅

⚠️ *Part of a stack created by [spr](https://github.com/ejoffe/spr). Do not merge manually using the UI - doing so may have unexpected results.*

### CODE REVIEW

File: scarb/src/lib.rs [hunk 1]
```diff
@@ -42,3 +42,4 @@ pub const TEST_ASSERTS_PLUGIN_NAME: &str = "assert_macros";
 pub const CAIRO_RUN_PLUGIN_NAME: &str = "cairo_run";
 pub const CARGO_MANIFEST_FILE_NAME: &str = "Cargo.toml";
 pub const CARGO_LOCK_FILE_NAME: &str = "Cargo.lock";
+pub const EXECUTABLE_PLUGIN_NAME: &str = "executable";
```
Comment (maciektr) [inline scarb/src/lib.rs:45]: Would you mind renaming the plugin to `cairo_execute`? I feel like it would be more idiomatic to our plugin names.

**Review by maciektr — APPROVED** [review]

File: scarb/src/lib.rs (hunk 1 above)
Comment (FroyaTheHen) [reply scarb/src/lib.rs:45]: not at all. just to be clear, you mean the value not the const
```suggestion
pub const EXECUTABLE_PLUGIN_NAME: &str = "cairo_execute";
```
??

Comment (maciektr) [reply scarb/src/lib.rs:45]: Yes, here and everywhere where it is needed (i.e. in scarblib templates).

File: scarb/scarblib/executable/Scarb.toml
Comment (maciektr) [inline scarb/scarblib/executable/Scarb.toml:1]: Pls change directory name as well :D

**Review by maciektr — APPROVED** [review]

File: utils/scarb-test-support/src/project_builder.rs [hunk 2]
```diff
@@ -114,6 +114,10 @@ impl ProjectBuilder {
         self.dep_builtin("starknet")
     }
 
+    pub fn dep_executable(self) -> Self {
```
Comment (Draggu) [inline utils/scarb-test-support/src/project_builder.rs:117]:
```suggestion
    pub fn dep_cairo_execute(self) -> Self {
```

**Review by maciektr — APPROVED** [review]

**Pull request merged**